src/components/calculator.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Display = (props) => {
+  const { value } = props;
+  return (
+    <div className="display">
+      <h1>{value}</h1>
+    </div>
+  );
+};
+
+Display.defaultProps = {
+  value: 0,
+};
+Display.propTypes = {
+  value: PropTypes.string,
+};
+
+const CalculatorButton = (props) => {
+  const { size, type, label } = props;
+  const classes = `${size} ${type} button`;
+  return (
+    <button type="button" className={classes}>{label}</button>
+  );
+};
+
+CalculatorButton.propTypes = {
+  size: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  label: PropTypes.string.isRequired,
+};
+
+const ButtonsContainer = () => (
+  <div className="buttonsContainer">
+    <CalculatorButton type="characterButton" size="s1" label="AC" />
+    <CalculatorButton type="characterButton" size="s1" label="+/-" />
+    <CalculatorButton type="characterButton" size="s1" label="%" />
+    <CalculatorButton type="operationButton" size="s1" label="÷" />
+    <CalculatorButton type="characterButton" size="s1" label="1" />
+    <CalculatorButton type="characterButton" size="s1" label="2" />
+    <CalculatorButton type="characterButton" size="s1" label="3" />
+    <CalculatorButton type="operationButton" size="s1" label="x" />
+    <CalculatorButton type="characterButton" size="s1" label="4" />
+    <CalculatorButton type="characterButton" size="s1" label="5" />
+    <CalculatorButton type="characterButton" size="s1" label="6" />
+    <CalculatorButton type="operationButton" size="s1" label="-" />
+    <CalculatorButton type="characterButton" size="s1" label="7" />
+    <CalculatorButton type="characterButton" size="s1" label="8" />
+    <CalculatorButton type="characterButton" size="s1" label="9" />
+    <CalculatorButton type="operationButton" size="s1" label="+" />
+    <CalculatorButton type="characterButton" size="s2" label="0" />
+    <CalculatorButton type="characterButton" size="s1" label="." />
+    <CalculatorButton type="operationButton" size="s1" label="=" />
+  </div>
+);
+
+const Calculator = () => (
+  <div className="calculatorContainer">
+    <Display />
+    <ButtonsContainer />
+  </div>
+);
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md#:~:text=Use%20class%20based%20components%20(All%20components%20should%20be%20class%2Dbased).), Please make the Calculator a class-based component instead of a functional component.